test/controllers/gems_controller_test.rb

```diff
@@ -12,4 +12,11 @@ def setup
     assert_template :index
     assert_select "title", "My Gems | RubyGems AdoptionCenter"
   end
+
+  test "should list gems" do
+    gems = return_gems
+
+    assert_equal Array, gems.class
+    assert_equal 'gem1', gems.first['name']
+  end
```

Comment:
Controller tests can be a bit confusing sometimes, especially in cases like this one when you're not really getting any obvious input from the user. I feel like testing `GemsController#index` doesn't really add any value.

It's probably a better idea to have an `integration` test check that this works as expected by rendering the list of gems that belong to a given user.